### PR TITLE
Add label feature

### PR DESCRIPTION
**Pull request by Krishanx92**
### Proposed changes in this pull request

Add label feature

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/APIAdmin.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
+/*
+*  Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (bhathiya) [inline components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/APIAdmin.java]: I think we no longer have the year in the license. Please double check.

File: components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/APIAdmin.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import org.wso2.carbon.apimgt.api.model.Label;
+
+import java.util.List;
+
+/**
+ * APIAdmin responsible for providing helper functionality
+ */
+public interface APIAdmin  {
+    /**
+     * Return labels specify by tenant
```
Comment (bhathiya) [inline components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/APIAdmin.java]: Let's change to "Returns labels of a given tenant"

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/LabelsApi.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+import org.wso2.carbon.apimgt.rest.api.admin.dto.LabelDTO;
+
+import java.util.List;
+
+import java.io.InputStream;
+import org.apache.cxf.jaxrs.ext.multipart.Attachment;
+import org.apache.cxf.jaxrs.ext.multipart.Multipart;
+
+import javax.ws.rs.core.Response;
+import javax.ws.rs.*;
```
Comment (hasuniea) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/LabelsApi.java]: Here avoid importing entire package.

Comment (Krishanx92) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/LabelsApi.java]: This is auto-gen code. can not change.

File: components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/APIAdmin.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
+/*
+*  Copyright (c) WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
[the remark this answers is not in the document]
Comment (Krishanx92) [reply components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/APIAdmin.java]: what is the issue here?

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java [hunk 5]
```diff
@@ -11262,4 +11263,200 @@ private String getUserIdFromConsumerKey (String consumerKey) throws APIManagemen
...
+             PreparedStatement statement = connection.prepareStatement(SQLConstants.ADD_LABEL_SQL)) {
+            try {
+                connection.setAutoCommit(false);
+
+                statement.setString(1, uuid);
+                statement.setString(2, label.getName());
+                statement.setString(3, label.getDescription());
+                statement.setString(4, tenantDomain);
+                statement.executeUpdate();
+                connection.commit();
```
Comment (bhathiya) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java]: Please move this commit to the bottom.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/LabelsApiService.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
+package org.wso2.carbon.apimgt.rest.api.admin;
+
+import org.wso2.carbon.apimgt.rest.api.admin.*;
```
[the remark this answers is not in the document]
Comment (Krishanx92) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/LabelsApiService.java]: This is auto-gen code. can not change.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/dto/LabelDTO.java [hunk 7]
```diff
@@ -0,0 +1,93 @@
+package org.wso2.carbon.apimgt.rest.api.admin.dto;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import io.swagger.annotations.*;
```
[the remark this answers is not in the document]
Comment (Krishanx92) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/dto/LabelDTO.java]: This is auto-gen code. can not change.

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/admin/site/themes/wso2/templates/label/label-add/js/edit.js [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+            isInvalidAttribute = true;
+            return false;
+        }
+    });
+    if (isInvalidAttribute) {
+        return false;
+    }
+
+    return true;
+}
```
Comment (isurulucky) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/admin/site/themes/wso2/templates/label/label-add/js/edit.js]: Lets get rid of all the extra new lines in this file.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/dto/LabelDTO.java [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+
+import javax.validation.constraints.NotNull;
+
+
+
+
+
+@ApiModel(description = "")
+public class LabelDTO  {
+
```
[the remark this answers is not in the document]
Comment (Krishanx92) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/dto/LabelDTO.java]: This is auto-gen code. can not change.

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/admin/site/themes/wso2/templates/label/label-add/js/edit.js (hunk 8 above)
Comment (Krishanx92) [reply features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/admin/site/themes/wso2/templates/label/label-add/js/edit.js]: Fix with https://github.com/wso2/carbon-apimgt/pull/5552

**Pull request merged**